api/v1/openapi.yaml

```diff
@@ -0,0 +1,953 @@
+          x-go-name: Invalid
+          schema:
+            "$ref": "#/definitions/Error"
+        '404':
+          description: Endpoint not found
+    delete:
+      summary: Delete endpoint
+      description: |
+        Deletes the endpoint specified by the ID. Deletion is imminent and
+        atomic, if the deletion request is valid and the endpoint exists,
+        deletion will occur even if errors are encounted in the process. If
+        errors have been encountered, the code 202 will be returned, otherwise
+        200 on success.
+
+        All resources associated with the endpoint will be freed and the
+        workload represented by the endpoint will be disconnected.It will no
+        longer be able to initiate or receive communications of any sort.
+      tags:
+      - endpoint
+      parameters:
+      - "$ref": "#/parameters/endpoint-id"
+      responses:
+        '200':
+          description: Success
+        '206':
+          description: Deleted with a number of errors encountered
+          x-go-name: Errors
+          schema:
+            type: integer
+        '400':
+          description: Invalid endpoint ID format for specified type. Details in error message"
+          x-go-name: Invalid
+          schema:
+            "$ref": "#/definitions/Error"
+        '404':
+          description: Endpoint not found
+  "/endpoint":
+    put:
+      summary: Create or update an endpoint
+      description: |
+        Updates an existing endpoint or creates it if it does not exist.
+        In order to update an endpoint, the ID field of the endpoint
+        must be set. The return code 200 and 201 indicates whether an
+        existing endpoint was updated or a new endpoint was created.
+
+        If an endpoint was updated and the change requires adaption of
+        the datapath, the relevant programs are automatically regenerated
+        and recompiled.
+      tags:
+      - endpoint
+      parameters:
+      - name: endpoint
+        in: body
+        required: true
+        schema:
+          "$ref": "#/definitions/Endpoint"
+      responses:
+        '200':
+          description: Updated
+        '201':
+          description: Created
+          schema:
+            "$ref": "#/definitions/EndpointResponse"
+        '400':
+          description: Invalid endpoint in request
+          x-go-name: Invalid
+          schema:
+            "$ref": "#/definitions/Error"
+        '460':
+          description: No endpoint information in request
+          x-go-name: Missing
+        '461':
+          description: Can only join existing endpoint in created state
+          x-go-name: InvalidState
+        '409':
+          description: Endpoint already exists
+          x-go-name: Exists
+        '500':
+          description: Join failed
+          x-go-name: Failed
+          schema:
+            "$ref": "#/definitions/Error"
+        '507':
+          description: Out of resources
+          x-go-name: OutOfResources
```

Comment:
Yes, this is not implemented yet but once we have it we can return a schema that represents this.